code/eyetracking/eyetrackingrun.py

```diff
@@ -0,0 +1,746 @@
+# Copyright 2023 The Axon Lab <theaxonlab@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+from pathlib import Path
+import string
+from typing import Optional, List, Tuple, Union
+import numpy as np
+import pandas as pd
+from scipy.signal import fftconvolve
+from pyedfread import edf, edfread
+import re
+import os
+import json
+import seaborn as sns
+import matplotlib.pyplot as plt
+from collections import defaultdict
+
+
+class EyeTrackingRun:
+    """
+
```

Comment:
Please write a docstring... Apologies I didn't have time to write it for you and just suggested placeholders.